### PR TITLE
GS-HW: Don't optimise clamp on shuffle

### LINKED ISSUE #5966: [BUG]: Time Crisis 2: Visual bugs
### Describe the Bug

For some reason, this game seems to have slight visual glitches across all of the backends. The Vulkan and OpenGL renderers have a noticeable white square that seems to stick around in the top left corner of the screen. DX11 and DX12 renderers have that, as well as the screen being a brighter white then usual. Software Renderer doesn't have either of the issues, but the demo play text looks garbled and warped.

### Reproduction Steps

Skip the intro and let the game go to demo play and it should be noticeable right away.

Use Vulkan and OpenGL for the white square at the top left.

Use DX11 and DX12 for the white square and a strangely brighter screen.

Use Software to see the demo play text garbled.

### Expected Behavior

There should be no white square at the top left of the screen during gameplay, weird white bright filter, or garbled demo play text.

### PCSX2 Revision

v1.7.2658

### Operating System

Windows 11

### If Linux - Specify Distro

_No response_

### CPU

Ryzen 5 5600H

### GPU

GTX 1650

### GS Settings

_No response_

### Emulation Settings

_No response_

### GS Window Screenshots

Vulkan and OpenGL Renderers:
Time Crisis Vulkan screen bug

DX11 and DX12 Renderers:
Time Crisis 2 DX12 screen bug

Software Renderer:
Time Crisis 2 Software screen bug

### Logs & Dumps

[Time Crisis 2 - Visual Bugs DX12.zip](https://github.com/PCSX2/pcsx2/files/8559616/Time.Crisis.2.-.Visual.Bugs.DX12.zip)
[Time Crisis 2 - Visual Bugs Software.zip](https://github.com/PCSX2/pcsx2/files/8559617/Time.Crisis.2.-.Visual.Bugs.Software.zip)
[Time Crisis 2 - Visual Bugs Vulkan.zip](https://github.com/PCSX2/pcsx2/files/8559619/Time.Crisis.2.-.Visual.Bugs.Vulkan.zip)

Tested with Vulkan, DX12, and Software renderers.

### PR DESCRIPTION
**Pull request by refractionpcsx2**
### Description of Changes
Don't optimise clamps on texture/channel shuffle

### Rationale behind Changes
Time Crisis 2 was a victim of this, the clamps were being optimised to only read 1 page of data, but it was a shuffle which we optimise to dot he whole screen, so it ended up shuffling everything in to the top left. So we need to check if this is a shuffle before we try to optimise the clamps.

### Suggested Testing Steps
True Time Crisis 2, especially the GS Dumps in the issue linked below.

Fixes #5966

Before:
image
image

After:
image
image

### CODE REVIEW

File: pcsx2/GS/Renderers/HW/GSRendererHW.cpp [hunk 1]
```diff
@@ -1543,12 +1585,14 @@ void GSRendererHW::Draw()
 		else if ((m_context->CLAMP.WMT & 2) && !(tmm.uses_boundary & TextureMinMaxResult::USES_BOUNDARY_V))
 			m_context->CLAMP.WMT = CLAMP_CLAMP;
 #endif
-
+		const int tw = 1 << TEX0.TW;
+		const int th = 1 << TEX0.TH;
+		const bool is_shuffle = m_channel_shuffle | m_texture_shuffle;
```
Comment (lightningterror) [inline pcsx2/GS/Renderers/HW/GSRendererHW.cpp:1590]: Should use logical ||.

Comment (refractionpcsx2) [reply pcsx2/GS/Renderers/HW/GSRendererHW.cpp:1590]: Why? both are boolean, it doesn't matter.

Comment (lightningterror) [reply pcsx2/GS/Renderers/HW/GSRendererHW.cpp:1590]: VS complains about it.

**Review by lightningterror — APPROVED** [review]
Dumps look good.

**Pull request merged**